sdk/core/Azure.Core.TestFramework/src/RecordedTestBaseOfTEnvironment.cs

```diff
@@ -23,5 +29,29 @@ public override void StartTestRecording()
         }
 
         public TEnvironment TestEnvironment { get; }
+
+        [OneTimeSetUp]
+        public async Task WaitForEnvironment()
```

Comment:
I wonder if we should move the implementation of this method onto `TestEnvironment` so even services that don't use RecordedTestBase can call it.